### PR TITLE
Fixes 1549

### LINKED ISSUE #1549: Qualitative metrics can be misleading with `beta_diversity`
When calling a qualitative metric, such as Jaccard, the input counts data are not automatically cast to presence/absence. The user may expect that to happen, and this can produce misleading results.

This issue was first opened on qiime2/q2-diversity#189. 

An example of the issue can be seen below. @gregcaporaso suggests two possible resolutions: (1) implementing the same steps as sklearn, or (2) updating skbio to call `sklearn.metrics.pairwise_distances` (plus new tests in skbio to ensure that we know if something changes in sklearn). 

Two near term work around options are to either cast input data to presence/absence prior to executing a qualitative metric, or to use `sklearn.metrics.pairwise_distances` as shown below.

The fast UniFrac methods are not impacted (e.g., 'unweighted_unifrac', and 'weighted_unifrac').

```python
import skbio
import sklearn
import numpy as np

>>> a = np.array([
    [42, 0, 37, 99, 1],
    [12, 1, 22, 88, 0],
    [25, 3, 23, 86, 0],
    [0, 0, 87, 12, 0]]
)

>>> dm1 = skbio.diversity.beta_diversity('jaccard', a, ids=['4ac2', 'e375', '4gd8', '9872'])
>>> dm2 = skbio.diversity.beta_diversity('jaccard', a > 0, ids=['4ac2', 'e375', '4gd8', '9872'])
>>> dm1 == dm2
False

>>> dm1 = skbio.diversity.beta_diversity('jaccard', a, ids=['4ac2', 'e375', '4gd8', '9872'], pairwise_func=sklearn.metrics.pairwise_distances)
>>> dm2 = skbio.diversity.beta_diversity('jaccard', a > 0, ids=['4ac2', 'e375', '4gd8', '9872'], pairwise_func=sklearn.metrics.pairwise_distances)
>>> dm1 == dm2
True
```

### PR DESCRIPTION
**Pull request by wasade**
Fixes #1549 

Please complete the following checklist:

* [X] I have read the guidelines in [CONTRIBUTING.md](https://github.com/biocore/scikit-bio/blob/master/CONTRIBUTING.md).

* [ ] I have documented all public-facing changes in [CHANGELOG.md](https://github.com/biocore/scikit-bio/blob/master/CHANGELOG.md).

* [ ] **This pull request includes code, documentation, or other content derived from external source(s).** If this is the case, ensure the external source's license is compatible with scikit-bio's [license](https://github.com/biocore/scikit-bio/blob/master/COPYING.txt). Include the license in the `licenses` directory and add a comment in the code giving proper attribution. Ensure any other requirements set forth by the license and/or author are satisfied. **It is your responsibility to disclose code, documentation, or other content derived from external source(s).** If you have questions about whether something can be included in the project or how to give proper attribution, include those questions in your pull request and a reviewer will assist you.

* [X] **This pull request does not include code, documentation, or other content derived from external source(s).**

**Note:** [REVIEWING.md](https://github.com/biocore/scikit-bio/blob/master/REVIEWING.md) may also be helpful to see some of the things code reviewers will be verifying when reviewing your pull request.

### CODE REVIEW

File: skbio/diversity/_driver.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 import itertools
 
 import numpy as np
-import scipy.spatial.distance
+import sklearn.metrics
```
Comment (RNAer) [inline skbio/diversity/_driver.py]: add sklearn added to documentation (and ci/xxx requirement files) as a dependency?

File: skbio/diversity/_driver.py [hunk 2]
```diff
@@ -342,6 +342,13 @@ def beta_diversity(metric, counts, ids=None, validate=True, pairwise_func=None,
     if validate:
         counts = _validate_counts_matrix(counts, ids=ids)
 
+    if 0 in counts.shape:
```
Comment (RNAer) [inline skbio/diversity/_driver.py]: add a test for this edge case?

Comment (wasade) [reply skbio/diversity/_driver.py]: it is already in `test_empty`

File: skbio/diversity/_driver.py (hunk 1 above)
Comment (wasade) [reply skbio/diversity/_driver.py]: it is already in ci, adding to docs
